### PR TITLE
2179: added functionality for submitManuscript to await on potential promise from updateManuscript

### LINKED ISSUE #2179: Investigate browser test failure in Happy Path
There seems to be an occasional browser test failure in the Happy Path browser test:

```
 ✖ Happy path (screenshots:
[2019-06-25T09:06:55.657Z]  /tmp/screenshots/2019-06-25_09-00-44/test-7/HeadlessChrome_75.0.3770_Linux_0.0.0/errors/1.png)
[2019-06-25T09:06:55.657Z] 
[2019-06-25T09:06:55.657Z]    1) AssertionError: expected 'Continue Submission' to deeply equal
[2019-06-25T09:06:55.657Z]       'Submitted'
[2019-06-25T09:06:55.657Z] 
[2019-06-25T09:06:55.657Z]       Browser: HeadlessChrome 75.0.3770 / Linux 0.0.0
[2019-06-25T09:06:55.657Z]       Screenshot:
[2019-06-25T09:06:55.657Z] 
[2019-06-25T09:06:55.657Z]    /tmp/screenshots/2019-06-25_09-00-44/test-7/HeadlessChrome_75.0.3770_Linux_0.0.0/errors/1.png
[2019-06-25T09:06:55.657Z] 
[2019-06-25T09:06:55.657Z]          122 |  await t
[2019-06-25T09:06:55.657Z]          123 |    .expect(dashboard.titles.textContent)
[2019-06-25T09:06:55.657Z]          124 |    .eql(manuscript.title)
[2019-06-25T09:06:55.657Z]          125 |    .expect(dashboard.statuses.textContent)
[2019-06-25T09:06:55.657Z]          126 |    // TODO this might cause a race condition
[2019-06-25T09:06:55.657Z]        > 127 |    .eql('Submitted')
[2019-06-25T09:06:55.657Z]          128 |})
[2019-06-25T09:06:55.657Z]          129 |
[2019-06-25T09:06:55.657Z]          130 |test('Ability to progress through the wizard is tied to
[2019-06-25T09:06:55.657Z]       validation', async t => {
[2019-06-25T09:06:55.657Z]          131 |  const navigationHelper = new NavigationHelper(t)
[2019-06-25T09:06:55.657Z]          132 |
[2019-06-25T09:06:55.657Z] 
[2019-06-25T09:06:55.657Z]          at eql (/home/xpub/test/submission.browser.js:127:6)
[2019-06-25T09:06:55.657Z] 
[2019-06-25T09:06:55.657Z] 
[2019-06-25T09:06:55.657Z]    2) Unhandled promise rejection:
[2019-06-25T09:06:55.657Z] 
[2019-06-25T09:06:55.657Z]       ExternalAssertionLibraryError { code: 'E53', isTestCafeError: true,
[2019-06-25T09:06:55.657Z]       callsite: CallsiteRecord { filename:
[2019-06-25T09:06:55.657Z]       '/home/xpub/test/submission.browser.js', lineNum: 126,
[2019-06-25T09:06:55.657Z]       callsiteFrameIdx: 6, stackFrames: [ [Object], [Object], [Object],
[2019-06-25T09:06:55.657Z]       [Object], [Object], [Object], [Object], CallSite {}, [Object],
[2019-06-25T09:06:55.657Z]       [Object], [Object], [Object], CallSite {} ], isV8Frames: true },
[2019-06-25T09:06:55.657Z]       errMsg: 'AssertionError: expected \'Continue Submission\' to deeply
[2019-06-25T09:06:55.657Z]       equal \'Submitted\'' }
[2019-06-25T09:06:55.657Z] 
[2019-06-25T09:06:55.657Z]       Browser: HeadlessChrome 75.0.3770 / Linux 0.0.0

```

Screenshot

### PR DESCRIPTION
**Pull request by erezmus**
#### Background

This PR should fix a potential race condition that occurs when submitting while an auto save is currently running, resulting in the front end having an out of date view of the submission

#### Any relevant tickets

Closes #2179

### CODE REVIEW

File: packages/component-submission/client/pages/SubmissionWizard.js [hunk 1]
```diff
@@ -78,15 +79,35 @@ export const SubmissionWizard = ({
...
+
+  // only submit once we are finished updating
+  // this is for the rare case that the form submit while its performing
+  // an auto save
   const handleSubmit = formValues =>
-    submitManuscript({
-      variables: { data: parseFormToOutputData(formValues) },
-    }).then(() => history.push(`/thankyou/${match.params.id}`))
+    updatePromise
+      .then(() =>
```
Comment (Cooryd) [inline packages/component-submission/client/pages/SubmissionWizard.js:105]: This probably shouldn't be addressed in this issue but this will likely mean we'll need to get a spinner on that submit button sooner rather then later @hdrury1

**Review by Cooryd — APPROVED** [review]
Code looks good, can we run this through browser tests a few times before merging though to catch any potential edge-cases?

**Review by will-byrne — APPROVED** [review]
